ui/tests/acceptance/sync/secrets/overview-test.js

```diff
@@ -76,12 +76,18 @@ module('Acceptance | sync | overview', function (hooks) {
         wasActivatePOSTCalled = true;
         return {};
       });
-
-      // override mirage to simulate no pre-existing destinations
-      this.server.get('/sys/sync/destinations', () => {});
     });
 
     test('the activation workflow works', async function (assert) {
+      assert.expect(8);
+
+      this.server.get('/sys/sync/destinations', () => {
+        assert.true(false, 'destinations is not called');
+      });
+      this.server.get('/sys/sync/associations', () => {
+        assert.true(false, 'associations is not called');
+      });
```

Comment:
this is a nit, but i'd recommend moving these assertions into their own tests to keep the scope of the tests smaller. i.e.

```typescript
module('when feature is activated', function (hooks) {

  test('it fetches destinations and associations', function() {
    // assert APIs are called
    // this.server.get('sys/sync/destinations', () ==> ...
  });

  module('when there are prexisting destinations', function() {
    test('it should transition to correct routes', function(){});
  })
});

module('when feature is not activated', function (hooks) {

  test('it does not fetch destinations and associations', function() {
    // assert APIs are not called
    // this.server.get('sys/sync/destinations', () ==> ...
  });

  test('the activation workflow works', function (){});
});

```

the benefits of keeping tests like these tightly scoped:
- the tests will be less likely to be affected by unrelated changes
- if a test fails, it's easier to pinpoint the issue and fix it. also, if you need to update the tests due to changes in the codebase later, it's easier to update smaller, more focused tests
- smaller tests typically run faster so you'll get faster feedback loops 🎉

let me know what you think!

Reply:
Agreed! Thanks for the details on why.